onnx/defs/text/defs.cc

```diff
@@ -0,0 +1,104 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "onnx/defs/schema.h"
+
+namespace ONNX_NAMESPACE {
+static const char* StringConcat_doc =
+    R"DOC(StringConcat concatenates string tensors elementwise (with NumPy-style broadcasting support))DOC";
+ONNX_OPERATOR_SET_SCHEMA(
```

Comment:
Do we need to handle inputs following the same description as the output of operator StringSplit?

Reply:
I wouldn't say that's necessary. We can transform the outputs from StringSplit pretty easily to do StringConcat. I quite like how simple this operator is right now.

Reply:
The corresponding operators in libraries like NumPy and Tensorflow don't behave like this (e.g. [np.char.add](https://numpy.org/doc/stable/reference/generated/numpy.char.add.html) vs [np.char.split](https://numpy.org/devdocs/reference/generated/numpy.char.split.html#numpy.char.split)) so I don't think there's a need.

Reply:
Agree, this binary version of concat looks good to me.